product_docs/docs/pgd/5.6/reference/functions.mdx

```diff
@@ -282,17 +281,60 @@ If a slot is dropped concurrently, the wait ends for that slot.
 If a node is currently down and isn't updating its slot, then the wait continues.
 You might want to set `statement_timeout` to complete earlier in that case.
 
+If you are using [Optimized Topology](../nodes/subscriber_only/optimizing-so), we recommend using [`bdr.wait_node_confirm_lsn`](/pgd/5.6/reference/functions#bdrwait_node_confirm_lsn) instead.
+)
+
 #### Synopsis
 
 ```sql
 bdr.wait_slot_confirm_lsn(slot_name text DEFAULT NULL, target_lsn pg_lsn DEFAULT NULL)
 ```
 
+#### Notes
+
+Requires `bdr_application` privileges to use.
+
+#### Parameters
+
+| Parameter    | Description                                                                 |
+|--------------|-----------------------------------------------------------------------------|
+| `slot_name`  | Name of the replication slot to wait for. If NULL, waits for all PGD slots. |
+| `target_lsn` | LSN to wait for. If NULL, uses the current write LSN on the local node.     |
+
+### `bdr.wait_node_confirm_lsn`
+
+Wait until a node passes a certain LSN.
+
+This function allows you to wait until the last write on this session was replayed to one or all nodes.
+
+Upon being called, the function waits for a node to pass a certain LSN.
+If no LSN is supplied, the current wal_flush_lsn (using the `pg_current_wal_flush_lsn()` function) position is used on the local node.
+Supplying a node name parameter tells the function to wait for that node to pass the LSN. 
+If no node name is supplied (by passing NULL), the function waits until all the nodes pass the LSN,
+
+The function operates by polling every 1000 ms for changes from other nodes. Those changes are detected either by looking up the `confirmed_flush_lsn` from the node's replication slot or by connecting to the node directly and querying the results of `pg_replication_origin_progress()`, passing the local node as origin.
+
+If you are confirming a specific node and the node is concurrently dropped, the wait for that node is ended. If a node is currently down, isn't updating or is simply not able to be connected to, the wait will continue indefinately. To avoid this, set the statement_timeout to the maximum amount of time you are prepared to wait.
```

Comment:
If changes are detected by looking up the `confirmed_flush_lsn` from the node's replication slot and the slot is dropped concurrently, the wait ends for that node. If changes are detected by connecting to the node directly and querying the results of `pg_replication_origin_progress()`, and the node is currently down, the wait will continue indefinately. To avoid this, set the statement_timeout to the maximum amount of time you are prepared to wait.

Reply:
I would not recommend documenting algorithm related details here. The querying of pg_replication_origin_progress() may change to use another API and we should not need to change this. Can we just say that with optimized topology a node may not have slots to all other nodes. Therefore, in these cases `bdr.wait_node_confirm_lsn` looks up the progress of the [origin](https://www.postgresql.org/docs/current/catalog-pg-replication-origin.html) on the named node or on all nodes if none given. When optimized topology is enabled, bdr.wait_slot_confirm_lsn may not work as desired, because all slots may not be present. But bdr.wait_node_confirm_lsn will work in all cases.

Reply:
I have adjusted with this in mind.